File: OOP based implmentation/OOP based implmentation/Project/login/CompanySignup.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "CompanySignup.h"

using namespace std;

CompanySignup::CompanySignup(const string &username, const string &password)
    : User(username, password, "company"), username(username), password(password)
{
}

void CompanySignup::saveToFile(const string &filename)
{
    ofstream writer(filename, ios_base::app);
    if (!writer.is_open())
    {
        cout << "===============================================================" << endl;
        cout << "An error occurred while saving company signup details." << endl;
        cout << "===============================================================" << endl;
        return;
    }

    writer << "UserType: " << this->userType << ", Username: " << this->username << ", Password: " << this->password << "\n";
    cout << "===============================================================" << endl;
    cout << "Company signup details saved successfully." << endl;
    cout << "===============================================================" << endl;
    writer.close();
}

File: OOP based implmentation/OOP based implmentation/Project/login/CompanySignup.java
package login;

import java.io.FileWriter;
import java.io.IOException;
import objs.user;

public class CompanySignup extends user {
    private String username;
    private String password;

    public CompanySignup(String username, String password) {
        super(username, password, "company");
        this.username = username;
        this.password = password;
    }

    public void saveToFile(String filename) {
        try (FileWriter writer = new FileWriter(filename, true)) {
            writer.write("UserType: " + this.userType + ", Username: " + this.username + ", Password: " + this.password + "\n");
            System.out.println("===============================================================");
            System.out.println("Company signup details saved successfully.");
            System.out.println("===============================================================");
        } catch (IOException e) {
            System.out.println("===============================================================");
            System.out.println("An error occurred while saving company signup details.");
            System.out.println("===============================================================");
            e.printStackTrace();
        }
    }
}
